CMakeLists.txt

```diff
@@ -84,6 +60,17 @@ if(NOT ENABLE_SHARED AND NOT ENABLE_STATIC)
     message(FATAL_ERROR "Both ENABLE_SHARED & ENABLE_STATIC are disabled. Need to enable at least one build flavour.")
 endif()
 
+# Set the C++ 17 standard. Have to use this method to make it consistent across the platforms
+# as cmake is doing funny things with the versions we have (including setting -std on MSVC).
```

Comment:
For windows and the version of cmake we are using, it definitely did the wrong thing and added -std:c++17 which doesn't pick up what we need (needs to be /std:c++17 from what I understand for MSVC). 

Otherwise I wanted for the other platforms/compilers to make it as close to MongoDB as possible. Apart from the tiered storage extension which is part of the shipping code, only test code uses C+ so we could argue that it doesn't really need to match but I think it is good to do it. This is why I have the extensions turned off.

Reply:
OK, that's interesting. Do you know which versions of CMake + MSVC you used? I read some documentation, and my understanding is that you need cmake >= 3.10 and MSVC >= 2017 for this to work. Maybe we need to clarify what the oldest supported MSVC is.

Regarding the C++ extensions, I didn't realise that MongoDB doesn't use them. If we're currently fine without extensions then yeah I agree it makes sense to stick to the standard.